test/controllers/drivers_controller_test.rb

```diff
@@ -3,173 +3,247 @@
-      # Arrange
       # Ensure that there is at least one Driver saved
-
-      # Act
-
-      # Assert
-
+      @driver.save
+      get drivers_path
+      must_respond_with :success
     end
 
     it "responds with success when there are no drivers saved" do
-      # Arrange
       # Ensure that there are zero drivers saved
-
-      # Act
-
-      # Assert
-
+      get drivers_path
+      must_respond_with :success
     end
   end
 
   describe "show" do
     it "responds with success when showing an existing valid driver" do
-      # Arrange
       # Ensure that there is a driver saved
-
-      # Act
-
-      # Assert
-
+      @driver.save
+      get driver_path(@driver.id)
+      must_respond_with :success
     end
 
     it "responds with 404 with an invalid driver id" do
-      # Arrange
       # Ensure that there is an id that points to no driver
-
-      # Act
-
-      # Assert
-
+      get driver_path(-1)
+      must_respond_with :not_found
     end
   end
 
   describe "new" do
     it "responds with success" do
+      get new_driver_path
+      must_respond_with :success
     end
   end
 
   describe "create" do
     it "can create a new driver with valid information accurately, and redirect" do
-      # Arrange
-      # Set up the form data
-
-      # Act-Assert
       # Ensure that there is a change of 1 in Driver.count
+      expect { 
+        post drivers_path, params: driver_hash
+      }.must_differ "Driver.count", 1
 
-      # Assert
       # Find the newly created Driver, and check that all its attributes match what was given in the form data
-      # Check that the controller redirected the user
+      expect(Driver.last.name).must_equal driver_hash[:driver][:name]
+      expect(Driver.last.vin).must_equal driver_hash[:driver][:vin]
 
+      # Check that the controller redirected the user
+      must_redirect_to driver_path(Driver.last.id)
     end
 
-    it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do
-      # Note: This will not pass until ActiveRecord Validations lesson
-      # Arrange
-      # Set up the form data so that it violates Driver validations
 
-      # Act-Assert
-      # Ensure that there is no change in Driver.count
+    it "does not create a driver if vin is not present, and responds with a redirect" do
+      driver_hash1 = {
+        driver: {
+          name: "Drivers are all Wonderful!"
+        }
+      }
+ 
+      expect {
+        post drivers_path, params: driver_hash1
+      }.wont_change "Driver.count"
 
-      # Assert
-      # Check that the controller redirects
+      assert_response :bad_request
+    end
 
+    it "does not create a driver if name is not present, and responds with a redirect" do
+      driver_hash2 = {
+        driver: {
+          vin: "12345678901234567"
+        }
+      }
+ 
+      expect {
+        post drivers_path, params: driver_hash2
+      }.wont_change "Driver.count"
```

Comment:
I think you may have changed something about the way that this route works that invalidated the tests here:

``` bash
Minitest::UnexpectedError:         NoMethodError: undefined method `capitalize' for nil:NilClass
            app/controllers/drivers_controller.rb:36:in `create'
            test/controllers/drivers_controller_test.rb:94:in `block (4 levels) in <main>'
            test/controllers/drivers_controller_test.rb:93:in `block (3 levels) in <main>'
```